lightning-graph-sync/src/processing.rs

```diff
@@ -0,0 +1,565 @@
+///
+/// `update_data`: `&[u8]` binary stream that comprises the update data
+pub fn update_network_graph(
+	network_graph: &network_graph::NetworkGraph,
+	update_data: &[u8],
+) -> Result<(), GraphSyncError> {
+	let mut read_cursor = io::Cursor::new(update_data);
+	read_network_graph(&network_graph, &mut read_cursor)
+}
+
+pub(crate) fn read_network_graph<R: Read>(
+	network_graph: &network_graph::NetworkGraph,
+	mut read_cursor: &mut R,
+) -> Result<(), GraphSyncError> {
+	let mut prefix = Vec::with_capacity(4);
+	prefix.resize(4, 0);
+	read_cursor.read_exact(&mut prefix)?;
+
+	let _deserialize_unsigned = match &prefix[..] {
+		&[76, 68, 75, 1] => true,
+		_ => {
+			return Err(GraphSyncError::ProcessingError(
+				"Prefix must equal 'LDK' and the version byte 1".to_string(),
+			));
+		}
+	};
+
+	// let chain_source: Option<Box<dyn lightning::chain::Access>> = None;
+	let current_timestamp = SystemTime::now()
+		.duration_since(UNIX_EPOCH)
+		.unwrap()
+		.as_secs() as u32;
+
+	let retrogressed_timestamp = current_timestamp - TIMESTAMP_RETROGRESSION_DELTA; // back-date updates by one week
+
+	let chain_hash: BlockHash = Readable::read(read_cursor)?;
+	let latest_seen_timestamp: u32 = Readable::read(read_cursor)?; //
+	let emulated_last_seen_timestamp = u32::max(latest_seen_timestamp, retrogressed_timestamp);
+
+	let node_id_count: u32 = Readable::read(read_cursor)?;
+	let mut node_ids: Vec<PublicKey> = Vec::with_capacity(node_id_count as usize);
+	for _ in 0..node_id_count {
+		let current_node_id = Readable::read(read_cursor)?;
+		node_ids.push(current_node_id);
+	}
+
+	let mut previous_scid = 0;
+	let announcement_count: u32 = Readable::read(read_cursor)?;
+	for _ in 0..announcement_count {
+		let features = Readable::read(read_cursor)?;
+
+		// handle SCID
+		let scid_delta: BigSize = Readable::read(read_cursor)?;
+		let short_channel_id = previous_scid + scid_delta.0;
+		previous_scid = short_channel_id;
+
+		let node_id_1_index: BigSize = Readable::read(read_cursor)?;
+		let node_id_2_index: BigSize = Readable::read(read_cursor)?;
+		let node_id_1 = node_ids[node_id_1_index.0 as usize];
+		let node_id_2 = node_ids[node_id_2_index.0 as usize];
+
+		let announcement_result = network_graph.update_channel_from_partial_announcement(
+			short_channel_id,
+			emulated_last_seen_timestamp as u64,
+			features,
+			node_id_1,
+			node_id_2,
+		);
+		if let Err(lightning_error) = announcement_result {
+			if let ErrorAction::IgnoreDuplicateGossip = lightning_error.action {
+				// everything is fine, just a duplicate channel announcement
+			} else {
+				return Err(lightning_error.into());
+			}
+		}
+	}
+
+	previous_scid = 0; // updates start at a new scid
+
+	let update_count: u32 = Readable::read(read_cursor)?;
+	if update_count < 1 {
+		return Ok(());
+	}
+	let non_incremental_update_presence_flag: u8 = Readable::read(read_cursor)?;
+	let has_non_incremental_updates = non_incremental_update_presence_flag == 1;
```

Comment:
Is it really worth saving the 26 static bytes for this? Lets simplify the code.

Reply:
It's actually a convenient simplification for the server not having to create a histogram of these values. I guess the server can just return some empty bytes here. Or I could abstract it away into a method that returns a defaults struct? 

But also happy to remove this code, too.

Reply:
Actually, I don't fully agree. The only code we're removing is the presence check, which is two lines. But we still have to read all the defaults after this.